File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1
        if nums[l] < nums[r]: return nums[l]
        num = nums[l]
        target = 1000
        while l <= r:
            mid = int((r + l) / 2)
            if nums[mid] < num:
                r = mid - 1
            elif nums[mid] >= num:
                l = mid + 1
            else:
                return nums[mid]
            target = min(target, nums[mid])
        return target

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cs
public class Solution {
    public int FindMin(int[] nums) {
        if (nums.Length == 1) return nums[0];
        else if (nums[0] < nums[^1]) return nums[0];
        int l = 0, r = nums.Length - 1, res = nums[0];
        while (l <= r)
        {
            if (nums[l] < nums[r]) 
            {
                res = Math.Min(res, nums[l]);
                break;
            }
            int mid = l + (r - l) / 2;
            res = Math.Min(res, nums[mid]);
            if (nums[mid] >= nums[l])
                l = mid + 1;
            else
                r = mid - 1;
        }
        return res;
    }
}
